hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/ReopenTableRegionsProcedure.java

```diff
@@ -92,8 +109,9 @@ protected Flow executeFromState(MasterProcedureEnv env, ReopenTableRegionsState
           LOG.info("Table {} is disabled, give up reopening its regions", tableName);
           return Flow.NO_MORE_STATE;
         }
-        regions =
-          env.getAssignmentManager().getRegionStates().getRegionsOfTableForReopen(tableName);
+        List<HRegionLocation> tableRegionsForReopen = env.getAssignmentManager()
+          .getRegionStates().getRegionsOfTableForReopen(tableName);
+        regions = prepareRegionsForReopen(tableRegionsForReopen);
```

Comment:
The name of the method is not self explaining. What prepare it is doing? This basically gets the Region locations of the mentioned region names right?

Reply:
You can even pass the regionNamesList also here.